learning_paths/api/v1/views.py

```diff
@@ -57,3 +64,56 @@ def get(self, request, learning_path_uuid):
         if serializer.is_valid():
             return Response(serializer.data, status=status.HTTP_200_OK)
         return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+
+class LearningPathUserGradeView(APIView):
+    """
+    API view to return the aggregate grade of a user in a learning path.
+    Only accessible by staff users.
+    """
+
+    permission_classes = (IsAuthenticated,)
+
+    def get(self, request, learning_path_uuid):
+        """
+        Fetch learning path grade
+        """
+        if not request.user.is_staff:
+            return Response(
+                {"detail": "Permission denied."}, status=status.HTTP_403_FORBIDDEN
+            )
+
+        username = request.query_params.get("username")
+        if not username:
+            return Response(
+                {"detail": "Username parameter is required."},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+        user = get_object_or_404(User, username=username)
+
+        learning_path = get_object_or_404(LearningPath, uuid=learning_path_uuid)
+
+        try:
+            grading_criteria = LearningPathGradingCriteria.objects.get(
+                learning_path=learning_path
+            )
+        except LearningPathGradingCriteria.DoesNotExist:
+            return Response(
+                {"detail": "Grading criteria not found for this learning path."},
+                status=status.HTTP_404_NOT_FOUND,
+            )
+
+        is_completed = is_learning_path_completed(user, learning_path)
```

Comment:
Shouldn't this be a part of the `LearningPathUserProgressView` endpoint?

Reply:
This was meant to show completion with respect to the completion threshold from the grading criteria. I've updated the variable name which should help make it clearer.

Reply:
@pkulkark, I'm still unsure why we want to show this here - it's the `LearningPathUserGradeView`, so it should stick to showing the grade-related data. The `LearningPathUserProgressView` shows the completion-related data. We have two options here:
1. Move all these fields to the `LearningPathUserProgressView` and remove `LearningPathUserGradeView` (which may be a good idea unless we plan to make these calculations much more complex).
2. Move `is_completion_threshold_met` to the `LearningPathUserProgressView` (though using separate endpoints may be harder to maintain once we start implementing the frontend part).

Reply:
I thought it would be useful to know if the completion threshold is met or not, in the api response, since we're setting the completion threshold as part of the grading criteria. I also wanted to separate out completion from progress. For example, if the course staff were to set the completion threshold as 50% and the progress is at 60% then the completion threshold is met but the user might still like to see their progress in the course.
I expect the progress api to be called more often, for example, if we're going to display a progress bar. Completion threshold being met seems to be irrelevant data there. What are your thoughts?

Reply:
@pkulkark, for the progress bar, we may eventually want to show the required completion somewhere. Therefore, we can use the second approach for now: remove the `is_completion_threshold_met` from this view and add `required_completion` to the information returned by the `LearningPathUserProgressView`.

Reply:
Makes sense. Updated it.